### PR TITLE
yt-dlp 2024.04.09

### PR DESCRIPTION
**Pull request by bevanjkay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Update was failing on `autobump` - https://github.com/Homebrew/homebrew-core/actions/runs/8631145141/job/23658941760#step:6:12666

### CODE REVIEW

File: Formula/y/yt-dlp.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@ class YtDlp < Formula
 
   desc "Feature-rich command-line audio/video downloader"
   homepage "https://github.com/yt-dlp/yt-dlp"
-  url "https://files.pythonhosted.org/packages/70/1d/f9617f151c124a15981cf79c47b6a53ac875942211b40010f0381296c51d/yt_dlp-2024.3.10.tar.gz"
-  version "2024.03.10"
-  sha256 "6e74cb14a69dbeb872c8ef4e0b8bbed2ee846ec633513cf3124a74c1faedc07b"
+  url "https://files.pythonhosted.org/packages/52/85/af44c30f0cc5d94ebbce365f8ee40eea6e9d7a1d26d50f16e3766ca8410c/yt_dlp-2024.4.9.tar.gz"
+  version "2024.04.09"
```
Comment (chenrui333) [inline Formula/y/yt-dlp.rb:7]: maybe remove this?
```suggestion
```

Comment (bevanjkay) [reply Formula/y/yt-dlp.rb:7]: I think it's required because the GitHub tag and the version on pypi differ.

Comment (chenrui333) [reply Formula/y/yt-dlp.rb:7]: where is the reference about such requirement?

---

I just did some local testing, it works fine removing it.

Comment (chenrui333) [reply Formula/y/yt-dlp.rb:7]: we also dont assert the version, even though after removing it, the version output is still `2024.04.09`

Comment (chenrui333) [reply Formula/y/yt-dlp.rb:7]: the version field is actually introduced by autobump, a bit weird, #165689

Comment (p-linnane) [reply Formula/y/yt-dlp.rb]: The filename in that PR doesn't have the leading 0 for the month, which is probably why.

**Review by p-linnane — APPROVED** [review]

**Pull request merged**